### PR TITLE
Waiting bug

### PR DESCRIPTION
**Pull request by enumag**
@P-Chang I think I might have found a bug in the waiting logic. In the case commited here the initial `Wait( 2000 );` seemd to work just fine. But all of the `Wait( 1000 );` calls seem to be ignored. I don't quite undestand why.

This sort of waiting is sometimes used in the script if there is a voice file that covers multiple sentences to try to better match the text and the voice. I tried to re-use that here because originallly this part used Line_WaitForInput instead which didn't look good in auto mode (the first part of the text waited for the entire voice).

Here is a save file near the location: https://www.dropbox.com/s/jni3km228uw0qhn/save058.dat?dl=0

### CODE REVIEW

**Comment by P-Chang** [comment]
[Location is https://github.com/07th-mod/tatarigoroshi/commit/370d217641d6830720a25fdbca9030002bfa3660

right?] edit

I missed  reference

**Comment by enumag** [comment]
yes

**Comment by P-Chang** [comment]
Probably it is related in text fade logic.

Line_* commands excluding Line_Continue set wait time that text fading time.
So, usually Wait(1000) command add wait time after text wait time.

In Line_Continue, wait time start with beginning of text fade. 

I may be able to add wait time in Line_Continue.

But normal Line_Continue setting have good things.We can set Wait() command same as voice flow.
However, game can adjust text fade time, so if user use longest setting, both way probably can't show text as good.

**Comment by enumag** [comment]
@P-Chang Afraid I understand very little from what you said. It's probably not worth explaining to me though. I trust you to decide what to do - even if your decision is to leave it as is to prevent breaking something else..

**Comment by P-Chang** [comment]
@enumag 
http://www.mediafire.com/file/ygvot6rdhz1du69/Chapter3Test.2017.10.8.rar

Probably it is fixed.
I decided add wait time for Line_Continue.

Although, it is not related in this.
Why did you increase "GCensorMaxNum" and "GCensor" in chapter1,2,4,5 ?
These don't have "GCensor" branch in current files.

And do you know good compressor for PNG ?

**Comment by enumag** [comment]
> Probably it is fixed.

Thank you! I'll try and let you know.

> Why did you increase "GCensorMaxNum" and "GCensor" in chapter1,2,4,5 ?

I didn't really think about that, I just took tata init file because it is the latest and used it everywhere. I don't think it's too important though. I mean sure I can revert it but if @ItaloKnox already made new archives it's not worth it to redo them again because of this.

> And do you know good compressor for PNG ?

Depends on if you want online compressor or installable one. And also whether you want to compress small images like icons or something large like PS3 backgrounds for Higurashi - many compressors don't work with big PNGs.

Generally I use [TinyPNG](https://tinypng.com/) or [CompressPNG](http://compresspng.com/). Beware though that some compressors are lossy even though they claim to be lossless - for example [compressor.io](https://compressor.io/).

For offline usage I use [PNGGauntlet](https://pnggauntlet.com/) but it does not work for large images. For batch compression of large images I wrote my own PHP script (it simply loads the image and saves it with max compression, nothing else - works fine for my needs).

**Comment by enumag** [comment]
@P-Chang It seems alright with the new DLL. Thanks for fixing it! Can you make this update for the other chapters as well?

**Comment by P-Chang** [comment]
@enumag 
Sure, I'll make it soon.

> it's not worth it to redo them again because of this.

I agree. I just wanted to know reason.

And thank you for telling me PNG things.

**Comment by P-Chang** [comment]
finished.
I also write it in update thread.

http://www.mediafire.com/file/4rt313dr8trawt4/chapter1dlls_v3XX.rar
http://www.mediafire.com/file/tihp2mlwqu10c1o/chapter2dlls_v2.x.x.rar
http://www.mediafire.com/file/ab9qvao3o3jp1b5/chapter3dlls_v3.x.x.rar
http://www.mediafire.com/file/tf014pqm25frrnh/chapter4dlls_v.2.x.x.rar
http://www.mediafire.com/file/z8xpnbzasjrdvgk/chapter5dlls_v2.x.x.rar

**Comment by enumag** [comment]
Great! Thanks. :-)

**Comment by enumag** [comment]
@P-Chang Please remind me... what exactly is the difference between Line_Continue and Line_ContinueAfterTyping? I need to write it down to the notes because now I'm not sure which one to use. I think one waits for voice and the other does not? anything else?

**Comment by P-Chang** [comment]
@enumag 
[Vanilla]
**Line_ContinueAfterTyping**
it add text waits

**Line_Continue**
It disable all voice waits in every frame until game use another Line_ command

[Patched]
**Line_ContinueAfterTyping**
it add text waits and if last Line_*  command is Line_Continue and game mode is auto, it will add voice wait.

**Line_Continue**
It disable all voice waits in every frame until game use another Line_ command
And I added text waits in it.



Normal voice wait is set by PlayVoice command if game mode is auto.
Basically two have no differences in normal mode now.
But you should use Line_ContinueAfterTyping in dialogs that not use long voice or dialogs is same as end of voice.

**Comment by enumag** [comment]
Hm... maybe Line_ContinueAfterTyping should always add voice wait? Or is there a reason against doing so?

Did you do some patching to the other Line_* commands? If so can you write that down as well?

Thank you!

**Comment by P-Chang** [comment]
> Did you do some patching to the other Line_* commands? If so can you write that down as well?

You mean Line_WaitForInput and Line_Normal can wait voice after Line_Continue? Yes, I patched it.
And I forget to write these things happen in only auto mode.

[Vanilla]
**Line_Normal**
add text waits and input wait and square marker, and deleting dialog after next input.

**Line_WaitForInput**
add text waits and input wait and triangle marker.

[Patched]
These work same as vanilla and if last Line_* command is Line_Continue and game mode is auto, it will add voice wait.


> Hm... maybe Line_ContinueAfterTyping should always add voice wait? Or is there a reason against doing so?

My patching is depend on Line_Continue is used in only long dialog with long voice.
Line_ContinueAfterTyping is used in many lines, so I want not to always add voice wait in it.

If MG use a lot of Line_Continue in reaming chapter, I need to remake logic.

**Comment by P-Chang** [comment]
I greped current files of chapter3.

Line that have Line_ContinueAfterTyping are 17015 and almost of these are name and line break.
Line that have Line_ContinueAfterTyping without name and line break are 646
Line that have Line_ContinueAfterTyping with `\\"` are 15

Probably I can add voice wait in Line_ContinueAfterTyping easily.
But I can't think it is good way. What's about your opinion of that?

Although we have variable that have last voice layer number, so I can make new modified `WaitForVoice() `command that can wait last layer automatically.

**Comment by enumag** [comment]
Basically I want to use it in cases where I want the next line to be played immediately after the previous without requiring user input (even when not using auto-mode). An example I found today is [here](https://github.com/07th-mod/tatarigoroshi/blob/master/Update/tata_010_02.txt#L1508-L1513).

However maybe it would interfere with names in non-auto-mode - possibly user input would not continue to next line while voice is playing. Maybe that could be fixed by using Line_Continue for names or something but I'm not sure.

Basically whatever you decide. Just tell me how to solve cases like the one I linked without risking the second line to start sooner than the first is finished in non-auto-mode. From your explanation above it seems that there is currently no other way than Line_WaitForInput (which is what was used in this case until I changed it today). If you think it's necessary and not worth fixing I'll revert it.

**Comment by P-Chang** [comment]
You dislike that using `WaitForVoice()`?  probably it is for this case.

If you dislike set layer number in that command, i can make new command easily.

**Comment by enumag** [comment]
No I just completely forgot about it. You're of course entirely correct. No need to change anything. Thanks for reminding me!
